### DISCUSSION TITLE
Streams override methods

### DISCUSSION THREAD

**Discussion by Adrian0012** [serverpod/serverpod#2945] [Q&A]
Reading the documentation i see that we can override some base methods
```
streamOpened is called when a user connects to a stream on the Endpoint.
streamClosed is called when a user disconnects from a stream on the Endpoint.
handleStreamMessage is called when a serialized message is received from a client.
```

However adding something as simple as a print statement in all of this methods yields nothing in the console. Is it something that I don't understand and as a result a print statement would not be shown? Or is it something else? Currently I am experimenting with streams and just wanted to see for myself when exactly this methods are being invoked

**Comment by hampuslavin** [comment]
Hi @Adrian0012! 👋 
Not to avoid your question, but have you checked out the new streaming _methods_ interface? https://docs.serverpod.dev/concepts/streams

It's a better and easier approach if you don't already have existing code using the old way. Let me know if you run into issues using those. And if you need to use the old way, of course we can continue debugging that issue.

**Reply by Adrian0012** [reply]
Hi @hampuslavin,
Thank you for your reply.
I am actually using as base the example from Pixorama project ( I assume that is a showcase of current best practices? )
I guess what I am trying to achieve is a way to identify how many clients are currently connected to the stream. Hence I was looking at streamOpened and streamClosed? For the purpose of my little experiment I only have a counter on the server-side nothing fancy :)

**Reply by Isakdl** [reply]
FYI depending on which repo you went to you may be looking at the old implementation.

Here is the one with the new streaming interface: https://github.com/serverpod/pixorama_live

**Reply by Adrian0012** [reply]
I have used the new interfaces ( [pixorama endpoint](pixorama_server/lib/src/endpoints/pixorama_endpoint.dart) )
From what I read the old interfaces required the user to manage the stream more closely whilst the new interfaces abstracts some of this management away from the user. However you could override some base methods from the old interfaces 
```
streamOpened is called when a user connects to a stream on the Endpoint.
streamClosed is called when a user disconnects from a stream on the Endpoint.
handleStreamMessage is called when a serialized message is received from a client.
```
Is there a way to access this using the new pattern?
